src/NuGet.Clients/VisualStudio/RestoreManager/IVsProjectRestoreInfo.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+
+namespace NuGet.VisualStudio
+{
+    /// <summary>
+    /// Contains complete project details needed for restore operation
+    /// </summary>
+    [ComImport]
+    [Guid("ab43992d-b977-436d-84c2-e76aeed20de2")]
+    public interface IVsProjectRestoreInfo
+    {
+        /// <summary>
+        /// Friendly project name.
+        /// </summary>
+        string ProjectName { get; }
+
+        /// <summary>
+        /// Name unique to the project across the solution.
+        /// </summary>
+        string ProjectUniqueName { get; }
+
+        /// <summary>
+        /// MSBuild project file path.
+        /// </summary>
+        string ProjectPath { get; }
+
+        /// <summary>
+        /// Assets file output path.
+        /// </summary>
+        string OutputPath { get; }
+
+        /// <summary>
+        /// User packages folder path.
+        /// </summary>
+        string PackagesPath { get; }
+
+        /// <summary>
+        /// Fallback folders.
+        /// </summary>
+        IReadOnlyList<string> FallbackFolders { get; }
+
+        /// <summary>
+        /// Target frameworks
+        /// </summary>
+        IReadOnlyList<FrameworkName> TargetFrameworks { get; }
+
+        /// <summary>
+        /// Project reference metadata. 
+        /// </summary>
+        IReadOnlyList<IVsProjectReference> ProjectReferences { get; }
+
+        /// <summary>
+        /// Package references.
+        /// </summary>
+        IReadOnlyList<IVsRestoreGraphItem> PackageReferences { get; }
+
+        /// <summary>
+        /// Framework assembly references.
+        /// </summary>
+        IReadOnlyList<IVsRestoreGraphItem> FrameworkAssemblies { get; }
```

Comment:
why there are 2 separate list of IVsRestoreGraphItem? PackageReferences itself will have target framework detail, wouldn't that be sufficient?